crates/astria-core/src/protocol/transaction/v1alpha1/action_group.rs

```diff
@@ -0,0 +1,491 @@
+    (FeeChangeAction, ActionGroup::BundlableSudo(BundlableSudo)),
+    (
+        FeeAssetChangeAction,
+        ActionGroup::BundlableSudo(BundlableSudo)
+    ),
+    (IbcRelay, ActionGroup::BundlableGeneral(BundlableGeneral)),
+);
+
+pub trait Group {
+    fn group(&self) -> ActionGroup;
+}
+
+impl Group for Action {
+    fn group(&self) -> ActionGroup {
+        match self {
+            Action::Sequence(_) => SequenceAction::GROUP,
+            Action::Transfer(_) => TransferAction::GROUP,
+            Action::ValidatorUpdate(_) => ValidatorUpdate::GROUP,
+            Action::SudoAddressChange(_) => SudoAddressChangeAction::GROUP,
+            Action::IbcRelayerChange(_) => IbcRelayerChangeAction::GROUP,
+            Action::Ics20Withdrawal(_) => Ics20Withdrawal::GROUP,
+            Action::InitBridgeAccount(_) => InitBridgeAccountAction::GROUP,
+            Action::BridgeLock(_) => BridgeLockAction::GROUP,
+            Action::BridgeUnlock(_) => BridgeUnlockAction::GROUP,
+            Action::BridgeSudoChange(_) => BridgeSudoChangeAction::GROUP,
+            Action::FeeChange(_) => FeeChangeAction::GROUP,
+            Action::FeeAssetChange(_) => FeeAssetChangeAction::GROUP,
+            Action::Ibc(_) => IbcRelay::GROUP,
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct BundlableGeneral;
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct General;
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct BundlableSudo;
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct Sudo;
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum ActionGroup {
+    BundlableGeneral(BundlableGeneral),
+    General(General),
+    BundlableSudo(BundlableSudo),
+    Sudo(Sudo),
+}
+
+impl fmt::Display for ActionGroup {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ActionGroup::BundlableGeneral(_) => write!(f, "BundlableGeneral"),
+            ActionGroup::General(_) => write!(f, "General"),
+            ActionGroup::BundlableSudo(_) => write!(f, "BundlableSudo"),
+            ActionGroup::Sudo(_) => write!(f, "Sudo"),
+        }
+    }
+}
+
+impl From<BundlableGeneral> for ActionGroup {
+    fn from(val: BundlableGeneral) -> ActionGroup {
+        ActionGroup::BundlableGeneral(val)
+    }
+}
+
+impl From<General> for ActionGroup {
+    fn from(val: General) -> ActionGroup {
+        ActionGroup::General(val)
+    }
+}
+
+impl From<BundlableSudo> for ActionGroup {
+    fn from(val: BundlableSudo) -> ActionGroup {
+        ActionGroup::BundlableSudo(val)
+    }
+}
+
+impl From<Sudo> for ActionGroup {
+    fn from(val: Sudo) -> ActionGroup {
+        ActionGroup::Sudo(val)
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+enum ErrorKind {
+    #[error("input contains mixed action types")]
+    Mixed,
+    #[error("attempted to create bundle with non bundleable `ActionGroup` type")]
+    NotBundleable,
+}
+
+#[derive(Debug)]
```

Comment:
For consistency with the rest of the crate:
```rust
#[derive(Debug, thiserror::Error)]
#[error(transparent)]
pub struct Error(ErrorKind);
```
And put the context into the kinds.